build_tools/benchmarks/post_benchmark_comment.py

```diff
@@ -0,0 +1,205 @@
+    }
+
+  def get(self, endpoint: str, payload: Any) -> requests.Response:
+    return requests.get(endpoint,
+                        data=json.dumps(payload),
+                        headers=self._api_headers)
+
+  def post(self, endpoint: str, payload: Any) -> requests.Response:
+    return requests.post(endpoint,
+                         data=json.dumps(payload),
+                         headers=self._api_headers)
+
+  def patch(self, endpoint: str, payload: Any) -> requests.Response:
+    return requests.patch(endpoint,
+                          data=json.dumps(payload),
+                          headers=self._api_headers)
+
+
+def post_to_gist(requester: APIRequester,
+                 filename: str,
+                 content: str,
+                 verbose: bool = False) -> str:
+  """Posts the given content to a new GitHub Gist and returns the URL to it."""
+
+  response = requester.post(endpoint=GITHUB_GIST_API,
+                            payload={
+                                "public": True,
+                                "files": {
+                                    filename: {
+                                        "content": content
+                                    }
+                                }
+                            })
+  if response.status_code != http.client.CREATED:
+    raise RuntimeError(
+        f"Failed to comment on GitHub; error code: {response.status_code}")
+
+  response = response.json()
+  if verbose:
+    print(f"Gist posting response: {response}")
+
+  if response["truncated"]:
+    raise RuntimeError(f"Content too large and gotten truncated")
+
+  return response["html_url"]
+
+
+def get_previous_comment_on_pr(requester: APIRequester,
+                               pr_number: int,
+                               gist_bot_user: str,
+                               comment_type_id: str,
+                               verbose: bool = False) -> Optional[int]:
+  """Gets the previous comment's id from GitHub."""
+
+  for page in range(1, MAX_PAGES_TO_SEARCH_PREVIOUS_COMMENT + 1):
+    response = requester.get(
+        endpoint=f"{GITHUB_IREE_API_PREFIX}/issues/{pr_number}/comments",
+        payload={
+            "per_page": 100,
+            "page": page,
+            "sort": "updated",
+            "direction": "desc"
+        })
+    if response.status_code != http.client.OK:
+      raise RuntimeError(
+          f"Failed to get PR comments from GitHub; error code: {response.status_code}"
+      )
+
+    comments = response.json()
+    if verbose:
+      print(f"Previous comment query response on page {page}: {comments}")
+
+    # Find the last comment from GITHUB_ACTIONS_USER and has the comment type id.
+    for comment in comments:
+      if (comment["user"]["login"] == gist_bot_user and
+          comment_type_id in comment["body"]):
+        return comment["id"]
+
+  return None
+
+
+def update_comment_on_pr(requester: APIRequester, comment_id: int,
```

Comment:
Nit: All of these methods take the `APIRequester` as an argument. What about just making them methods on it? Or if you want to keep the requester scoped, you could compose and have another class that is initialized with a requester.

Reply:
I have `APIRequester` as a standalone class because it makes it easier to swap that with a fake/mock object in the unit tests.

Bundled all these methods into the class `GithubClient`, which takes `APIRequester` in the contrustion.